wallets/provider-walletconnect-2/src/index.ts

```diff
@@ -176,26 +163,33 @@ export const switchNetwork: SwitchNetwork = async ({
   instance,
   meta,
   getState,
+  updateChainId,
 }) => {
-  const needRecreateSession = needSessionRecreateOnSwitchNetwork(instance);
-  config.isAsyncSwitchNetwork = true;
-  if (needRecreateSession) {
-    config.isAsyncSwitchNetwork = false;
-    /**
-     * In case of trust wallet that doesn't support switch chain method,
-     * we need to handle it manually by deleting last session and create a new one
-     * with correct chain id.
-     */
-    const session = await trySwitchByCreatingNewSession(instance, {
-      network,
-      meta,
-    });
-    instance.session = session;
+  const evm = evmBlockchains(meta);
+  const chainId = evm.find((chain) => chain.name === network)?.chainId || '0x1';
+  const chaindIdStr = new ChainId({
+    namespace: NAMESPACES.ETHEREUM,
+    reference: String(parseInt(chainId)),
+  }).toString();
+
+  const session = instance.session;
+  const evmNamespace = session.namespaces[NAMESPACES.ETHEREUM];
+  const authorizedChains = evmNamespace?.chains || [];
+  const authorizedMethods = evmNamespace?.methods || [];
+
+  if (
+    authorizedMethods.includes(EthereumRPCMethods.SWITCH_CHAIN) &&
+    !ignoreNamespaceMethods(instance)
+  ) {
+    const currentNetwork = getState?.().network || Networks.ETHEREUM;
+    await updateSessionAccounts(instance, network, currentNetwork, meta);
+    await switchOrAddEvmChain(meta, network, currentNetwork, instance);
+  } else if (authorizedChains.includes(chaindIdStr)) {
+    updateChainId(chainId);
     return;
+  } else {
+    throw Error(`Chain ${chainId} is not configured.`);
```

Comment:
you can use our new logging packages to report the error. `import {error} from "@rango-dev/logging-core"`

Reply:
Please use `new` before `Error`.